### ISSUE TITLE
feature: selective prediction metric

### ISSUE THREAD

**Issue by Maitreyapatel** [Maitreyapatel/reliability-checklist#45]
**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
An evaluation skim like other existing evals. 

**Describe alternatives you've considered**
We need to manually modify the code to get logits, which is not suggested.

**Additional context**
N/A
